### PR TITLE
[New RTA] Input Capture via Keylog

### PR DESCRIPTION
**Pull request by Samirbous**
APIs in scope covered by 2 seperate RTAs  :

SetWindowsHookEx (`collection_keylog_hook_keystate`) 
GetAsyncKeyState (`collection_keylog_hook_keystate`) 
RegisterRawInputDevices (`collection_keylog_rawinputdevice`)

image

### CODE REVIEW

File: rta/collection_keylog_rawinputdevice.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    endpoint=[
+        {"rule_name": "Keystroke Input Capture via DirectInput", "rule_id": "102b5c1a-7f2a-4254-8b26-6b299705fce7"},
+        {"rule_name": "Keystroke Input Capture via RegisterRawInputDevices", "rule_id": "4dbb9dfb-b3e2-49d7-8919-d6f221526df4"},
+    ],
+    siem=[],
+    techniques=["T1056", "T1056.001"],
+)
+
+
+@common.requires_os(common.WINDOWS)
```
Comment (Mikaayenson) [inline rta/collection_keylog_rawinputdevice.py:24]:
```suggestion
@common.requires_os(*metadata.platforms)
```

File: rta/collection_keylog_hook_keystate.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    global hHook
+    user32 = ctypes.windll.user32
+
+    if code == HC_ACTION and w_param == WM_KEYDOWN:
+        print("Key down")
+
+    return user32.CallNextHookEx(hHook, code, w_param, l_paraml)
+
+
+@common.requires_os(common.WINDOWS)
```
Comment (Mikaayenson) [inline rta/collection_keylog_hook_keystate.py:66]:
```suggestion
@common.requires_os(*metadata.platforms)
```

**Review by Mikaayenson — APPROVED** [review]
@Samirbous this LGTM. Ideally these would auto PR from react, but so that we dont leave this open any longer, we can migrate these back later.

**Review by eric-forte-elastic — APPROVED** [review]
Concur with Mika's comment. Will need reformatting/linting going into REACT, but that will sync back here as well. Functionality looks good :+1:

**Pull request merged**